core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -1178,6 +1178,10 @@ final void putAll(Map<? extends K, ? extends V> map, Metadata metadata, long exp
    private void putAllInternal(Map<? extends K, ? extends V> map, Metadata metadata, long explicitFlags, InvocationContext ctx) {
       InfinispanCollections.assertNotNullEntries(map, "map");
       Metadata merged = applyDefaultMetadata(metadata);
+      // Vanilla PutMapCommand returns previous values; add IGNORE_RETURN_VALUES as the API will drop the return value
+      // interceptors are free to clear this flag if appropriate (since interceptors are the only consumers of the
+      // return value)
```

Comment:
I agree a new flag wouldn't make sense, I was actually thinking of a boolean in the command itself, like `isForwarded`. My reasoning was that it would be easier to add something that's specifically internal-only than fixing all the places in the code that ignore `IGNORE_RETURN_VALUES` for `PutMapCommand`. E.g. should `BaseTypeConverterInterceptor` unwrap the results?